plugins/plugin-ssh-key/che-plugin-ssh-key-server/src/test/java/org/eclipse/che/plugin/ssh/key/HttpSshServiceClientTest.java

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.ssh.key;
+
+import org.eclipse.che.api.core.rest.HttpJsonRequest;
+import org.eclipse.che.api.core.rest.HttpJsonRequestFactory;
+import org.eclipse.che.api.core.rest.HttpJsonResponse;
+import org.eclipse.che.api.ssh.server.SshService;
+import org.eclipse.che.api.ssh.shared.dto.GenerateSshPairRequest;
+import org.eclipse.che.api.ssh.shared.dto.SshPairDto;
+import org.mockito.Mock;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+import static javax.ws.rs.core.UriBuilder.fromUri;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for {@link HttpSshServiceClient}
+ *
+ * @author Igor Vinokur
+ */
+@Listeners(MockitoTestNGListener.class)
+public class HttpSshServiceClientTest {
+    private final static String SSH_KEY_SERVICE = "service";
+    private final static String SSH_SERVICE_URL = "apiUrl/ssh";
+    private final static String SSH_KEY_NAME    = "name";
+    private final static String API_URL         = "apiUrl";
+
+    @Mock
+    private HttpJsonRequestFactory requestFactory;
+    @Mock
+    private HttpJsonRequest        jsonRequest;
+    @Mock
+    private HttpJsonResponse       jsonResponse;
+
+    private HttpSshServiceClient client;
+
+    @BeforeMethod
+    public void setup() throws Exception {
+        when(jsonRequest.usePostMethod()).thenReturn(jsonRequest);
```

Comment:
You can create mock mannually with SelfReturningAnswer instead of this code. To wit `jsonRequest = mock(HttpJsonRequest.class, new SelfReturningAnswer());`
P.S. SelfReturningAnswer is from org.eclipse.che.core:che-core-commons-test